internal/worker/json.go

```diff
@@ -23,12 +24,14 @@ type OSBuildJob struct {
 }
 
 type OSBuildJobResult struct {
-	Success       bool                   `json:"success"`
-	OSBuildOutput *osbuild.Result        `json:"osbuild_output,omitempty"`
-	TargetResults []*target.TargetResult `json:"target_results,omitempty"`
-	TargetErrors  []string               `json:"target_errors,omitempty"`
-	UploadStatus  string                 `json:"upload_status"`
-	PipelineNames *PipelineNames         `json:"pipeline_names,omitempty"`
+	Success         bool                   `json:"success"`
+	OSBuildOutput   *osbuild.Result        `json:"osbuild_output,omitempty"`
+	TargetResults   []*target.TargetResult `json:"target_results,omitempty"`
+	TargetErrors    []string               `json:"target_errors,omitempty"`
+	TargetJobErrors []clienterrors.Error   `json:"target_job_errors,omitempty"`
+	JobError        *clienterrors.Error    `json:"job_error"`
```

Comment:
Is this for backwards compatibility?  Adding fields to the existing result object instead of modifying?
It's getting a bit messy with the double result checking of the result in the worker, but maybe this is necessary...

Reply:
Yeah my thoughts there were to not break backwards compatibility, but happy enough to modify it instead, if we are happy to do that.

Reply:
Or like you say, return an empty result and check in composer instead.

Reply:
It looks like the only time we read the results object in composer for errors is in the Koji api  i.e. `if initResult.KojiError != ""`) For the rest it's only ever happening in the workers.